### PR TITLE
Remove More Optional Class Testing

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR follows #2094 in gutting more of the Ice/optional tests.
This is the penultimate PR, since we're there's still some optional classes being passed around via the streaming APIs,
and the Java tests in particular can use some serious cleaning up since 'java:optional' is dead metadata.

It's mostly just removal, so feel free to only look at the languages you care about.

Anyways, this PR:
- removes `supportsNullOptional`
- removes our `opMG` tests which test the interplay of `marshaled-result` with optional classes
- removes optional class fields we had stuck in some exceptions
- simplifies (but keeps) the 'opOneOptional' test.
It's a simple test, and believe-it-or-not, it's the only test I see that ensures optional data members survive a round trip from client->server->client, which seems worth keeping a test for.

### CODE REVIEW

File: cpp/test/Ice/logger/log.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+-- 05/08/24 16:59:34115 aplicación: info: XXX
```
Comment (bernardnormier) [inline cpp/test/Ice/logger/log.txt:1]: It looks like this file was added by accident.

File: cpp/test/Ice/optional/Test.ice [hunk 2]
```diff
@@ -294,21 +291,11 @@ interface Initial
...
 
-    ["marshaled-result"] optional(1) G opMG1();
-    ["marshaled-result"] optional(1) G opMG2(optional(2) G p1, out optional(3) G p2);
-
     bool supportsRequiredParams();
 
     bool supportsJavaSerializable();
 
     bool supportsCsharpSerializable();
-
```
Comment (bernardnormier) [inline cpp/test/Ice/optional/Test.ice:305]: Should also remove supportsCsharpSerialize above. We dropped support for C# serialization a while ago.

Comment (bernardnormier) [reply cpp/test/Ice/optional/Test.ice:305]: and supportsRequiredParams. I can't find any call to this operation.

**Review by bernardnormier — APPROVED** [review]
It would be good to reenable the cross testing with C#.

See ice/.github/workflows/ci.yml

File: cpp/test/Ice/optional/Test.ice (hunk 2 above)
Comment (InsertCreativityHere) [reply cpp/test/Ice/optional/Test.ice:305]: Yes, this is _only_ of the Java mapping.
Java is the only language which returns `true` for this, and it's also the only language that calls it.
At the top of `Ice/Optional/AllTests.java`

That's part of the Java cleanup I mentioned in the top description.

File: cpp/test/Ice/logger/log.txt (hunk 1 above)
Comment (InsertCreativityHere) [reply cpp/test/Ice/logger/log.txt:1]: Good catch. I was committing too hastily.
Removed.

**Review by pepone — APPROVED** [review]

File: cpp/test/Ice/logger/log.txt (hunk 1 above)
Comment (ReeceHumphreys) [reply cpp/test/Ice/logger/log.txt]: Maybe we should update the `.gitignore` to catch this file if it was generated.

**Review by ReeceHumphreys — APPROVED** [review]
Looked at the JS and Python and it looked good

**Review by externl — APPROVED** [review]

**Pull request merged**